Sources/Topsort-Analytics/Models/Auctions.swift

```diff
@@ -0,0 +1,135 @@
+import Foundation
+
+// Auctions requests models
+
+public struct AuctionGeoTargeting: Codable {
+    /**
+     * The location this auction is being run for.
+     */
+    let location: String
+
+    public init(location: String) {
+        self.location = location
+    }
+}
+
+public struct AuctionCategory: Codable {
+    /**
+     * The category ID of the bids that will participate in an auction.
+     */
+    let id: [String]?
+
+    /**
+    * An array containing the category IDs of the bids that will participate in an auction.
+    * In order to participate in an auction, a bid product must belong to all of the categories provided in the auction request.
+    */
+    let ids: [String]?
+
+    /**
+    * An array of disjunctions.
+    * In order to participate in an auction, a bid product must belong to at least one of the categories for each of the disjunctions provided in the auction request.
+    */
+    let disjunctions: [String]?
+
+    public init(id: [String]? = nil, ids: [String]? = nil, disjunctions: [String]? = nil) {
+        self.id = id
+        self.ids = ids
+        self.disjunctions = disjunctions
+    }
+}
+
+public struct AuctionProducts: Codable {
+    /**
+    * The list of product IDs to be considered in the auction.
+    */
+    let ids: [String]
+
+    /**
+    * The list of quality scores for the products in the auction.
+    */
+    let qualityScores: [Double]?
+
+    public init(ids: [String], qualityScores: [Double]? = nil) {
+        self.ids = ids
+        self.qualityScores = qualityScores
+    }
+}
+
+public struct Auction: Codable {
```

Comment:
Instead of having one model for both listings and banners, create one model for each and use the `type` as the discriminator.

Reply:
Initially, I had done it this way, but I decided to do it together for two reasons:
- The API receives an array with everything together, and Swift did not supported me creating an array with those two different types.
- Since the order in which the auctions are sent matters, sending banners and listings separately could cause confusion. The client would then have to check the response to see the order we chose internally.

Let me know what you think.

Reply:
It all comes to a bad API :(